sycl/unittests/kernel-and-program/MultipleDevsCache.cpp

```diff
@@ -109,13 +156,14 @@ TEST_F(MultipleDeviceCacheTest, ProgramRetain) {
     sycl::queue Queue(Context, Devices[0]);
     assert(Devices.size() == 2 && Context.get_devices().size() == 2);
 
-    auto KernelID = sycl::get_kernel_id<TestKernel<>>();
+    auto KernelID = sycl::get_kernel_id<MultipleDevsCacheTestKernel>();
     auto Bundle = sycl::get_kernel_bundle<sycl::bundle_state::input>(
         Queue.get_context(), {KernelID});
     assert(Bundle.get_devices().size() == 2);
 
-    Queue.submit(
-        [&](sycl::handler &cgh) { cgh.single_task<TestKernel<>>([]() {}); });
+    Queue.submit([&](sycl::handler &cgh) {
+      cgh.single_task<MultipleDevsCacheTestKernel>([]() {});
+    });
```

Comment:
Suggest add an assert which check for number of device images in the kernel bundle to avoid this problem in future. E.g.
```
EXPECT_EQ(BundleImpl->size(), 1);
```

Reply:
Good shout! It has been added.